src/js/actions/documents.js

```diff
@@ -465,17 +465,16 @@ define(function (require, exports) {
                     recentFilesPromise = this.transfer(application.updateRecentFiles),
                     updateTransformPromise = this.transfer(ui.updateTransform),
                     deleteTempFilesPromise = this.transfer(libraryActions.deleteGraphicTempFiles,
-                        documentID, isDocumentSaved);
+                        documentID, isDocumentSaved),
+                    policyPromise = newDocument ? Promise.resolve() :
+                        this.transfer(toolActions.resetBorderPolicies);
 
                 return Promise.join(resetLinkedPromise,
                         resetHistoryPromise,
                         updateTransformPromise,
                         recentFilesPromise,
-                        deleteTempFilesPromise);
-            })
-            .then(function () {
-                // This must run after the above transfers. See #2785.
-                return this.transfer(toolActions.resetBorderPolicies);
+                        deleteTempFilesPromise,
+                        policyPromise);
```

Comment:
That should be OK. The transfer queue will ensure that the policy transfer doesn't run at the same time with anything else that changes the policies. I moved it previously (and added that comment) to try to clarify that running it concurrently with the other reset actions would cause an atomicity violation. But, with the transfer queue that should no longer be necessary.